lib/api/transactions.js

```diff
@@ -450,51 +468,42 @@ module.exports = function (app) {
 	};
 
 	this.getTransactionsByAddress = function (query, error, success) {
-		const data = normalizeTransactionParams(query, error);
-		if (typeof data === 'string') {
-			return error({ success: false, error: data });
+		const queryList = normalizeTransactionParams(query, error);
+		if (typeof queryList === 'string') {
+			return error({ success: false, error: queryList });
 		}
 
-		const indexOfById = (list, item) => {
-			let index = -1;
-			list.forEach((mem, i) => {
-				if (mem.id === item.id) {
-					index = i;
+		const queryCallsList = queryList.map(directionQuery => (callback) => {
+			request.get({
```

Comment:
I think we cannot use pure ES Array.map() and an asynchronous operation there. I would suggest replacing it with async.map or similar solution based on async/Promise.

Reply:
The Array.map() here is only generating an array of functions to be later executed on async.parallel.

Reply:
It's ok then.